src/util/sighash.rs

```diff
@@ -172,7 +197,49 @@ impl<'s> ScriptPath<'s> {
     }
 }
 
-impl<R: Deref<Target=Transaction>> SigHashCache<R> {
+impl From<LegacySigHashType> for SigHashType {
+    fn from(s: LegacySigHashType) -> Self {
+        match s {
+            LegacySigHashType::All => SigHashType::All,
+            LegacySigHashType::None => SigHashType::None,
+            LegacySigHashType::Single => SigHashType::Single,
+            LegacySigHashType::AllPlusAnyoneCanPay => SigHashType::AllPlusAnyoneCanPay,
+            LegacySigHashType::NonePlusAnyoneCanPay => SigHashType::NonePlusAnyoneCanPay,
+            LegacySigHashType::SinglePlusAnyoneCanPay => SigHashType::SinglePlusAnyoneCanPay,
+        }
+    }
+}
+
+impl From<SigHashType> for LegacySigHashType {
+    fn from(s: SigHashType) -> Self {
+        match s {
+            SigHashType::Default => LegacySigHashType::All,
+            SigHashType::All => LegacySigHashType::All,
+            SigHashType::None => LegacySigHashType::None,
+            SigHashType::Single => LegacySigHashType::Single,
+            SigHashType::AllPlusAnyoneCanPay => LegacySigHashType::AllPlusAnyoneCanPay,
+            SigHashType::NonePlusAnyoneCanPay => LegacySigHashType::NonePlusAnyoneCanPay,
+            SigHashType::SinglePlusAnyoneCanPay => LegacySigHashType::SinglePlusAnyoneCanPay,
+        }
+    }
+}
+
+impl SigHashType {
+    /// Break the sighash flag into the "real" sighash flag and the ANYONECANPAY boolean
+    pub(crate) fn split_anyonecanpay_flag(self) -> (SigHashType, bool) {
+        match self {
+            SigHashType::Default => (SigHashType::Default, false),
```

Comment:
We should fail when the user provides `Default` for segwitv0 sighash

Reply:
this is for the taproot sighash, is it correct?

Reply:
Yeah sorry. I don't know why this comment is placed for this code block. But I wanted is fixed in 993dae01b35038cc49e216afff9525d1b870f419